xla/literal.h

```diff
@@ -360,12 +360,16 @@ class LiteralBase {
   // function which would lead to inconsistencies. Use Hash instead.
   template <typename H>
   friend H AbslHashValue(H state, const LiteralBase& value) {
+#if (!defined(__GNUC__)) // clang
     static_assert(false,
```

Comment:
can you instead of disabling this check for gcc change it to `static_assert(sizeof(H) == 0, ...` (see https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2593r0.html)

Reply:
thanks for this link! looks `static_assert(always_false<T>::value);` is a more acceptable way... :)